Update Obvious Fix to require "@pivotal-issuemaster this is an obvious fix"

simonbasle opened the pull request:
Had to bump Gradle to 4.8.1 to make it compatible with latest IntelliJ, based on Java11.

simonbasle committed:
Polish build: bump to Gradle 4.8.1 for compatibility with Java 10+ IDEs

simonbasle committed:
Ignore the Obvious Fix words if the bot is not mentioned in comments

This commit helps with "false positives" whenever people use the words
"Obvious Fix" in a comment that was not addressed to the bot.

For example, one could ask the maintainers if the PR "can be considered
an obvious fix". This sort of comment would previously trigger the bot
to answer "this is an obvious fix", misleading the user into thinking no
CLA signature is needed.

Now such a comment will be taken into consideration if and only if the
comment starts with a mention to the bot in the `@claUserLogin ` format.

Fixes #150

rwinch commented on src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java:
```diff
@@ -175,6 +177,7 @@ private boolean hasObviousFixComment(Collection<? extends Comment> comments, Str
 		Optional<? extends Comment> obviousFixComment = comments.stream() //
 				.filter(comment -> comment.getUser() != null && !claUserLogin.equals(comment.getUser().getLogin())) //
 				.filter(comment -> hasObviousFix(comment.getBody())) //
+				.filter(comment -> comment.getBody().startsWith("@" + claUserLogin + " ")) //
```
I wonder if it makes more sense to update `hasObviousFix` method rather than adding an additional filter here?

simonbasle replied:
I started by doing that, but then this method also covers the PR body, which is a bit of a different case in my opinion since it is less likely the author would know to mention the bot from the PR's inception. But as I pondered in the issue, I wonder if it would make sense to stop watching the PR body altogether.

rwinch replied:
I think it does make sense as we often get repeat contributors that might include it in the body. Either way, if they mention the bot and state it is an obvious fix it seems pretty clear the intention.

Also...I think that that the mention should not need be the start of the text but the mention should be next to the obvious fix text

simonbasle replied:
so explicitly detect the string `"@bot this is an obvious fix"`?

rwinch replied:
Yes. I think so

rwinch reviewed (changes requested):
Thanks for the PR @simonbasle! I have provided feedback inline.

simonbasle committed:
Stricter obvious fix trigger sentence, but accepted anywhere in comment

This includes the body of the PR and any comments. Format is
`@bot this is an obvious fix`.
